src/test/java/no/unit/nva/cristin/projects/FetchCristinProjectsTest.java

```diff
@@ -232,8 +226,54 @@ void handlerReturnsProjectsWrapperWithAllMetadataButEmptyHitsArrayWhenNoMatchesA
 
     @Test
     void getsCorrectUriWhenCallingQueryProjectsUriBuilder() throws Exception {
-        assertEquals(new URI(GET_CRISTIN_PROJECTS_EXAMPLE_URI), cristinApiClientStub
-            .generateQueryProjectsUrl(Map.of(TITLE, RANDOM_TITLE, CRISTIN_LANGUAGE_PARAM, LANGUAGE_NB)));
+        assertEquals(new URI(QUERY_CRISTIN_PROJECTS_EXAMPLE_URI), cristinApiClientStub
+            .generateQueryProjectsUrl(Map.of(TITLE, RANDOM_TITLE, LANGUAGE_KEY, LANGUAGE_NB)));
+    }
+
+    @Test
+    void handlerReturnsServerErrorExceptionWhenBackendThrowsGenericException() throws Exception {
+        cristinApiClientStub = spy(cristinApiClientStub);
+
+        doThrow(RuntimeException.class).when(cristinApiClientStub).generateQueryProjectsUrl(any());
+        handler = new FetchCristinProjects(cristinApiClientStub, environment);
+        GatewayResponse<ProjectsWrapper> response = sendDefaultQuery();
+
+        assertEquals(HttpURLConnection.HTTP_INTERNAL_ERROR, response.getStatusCode());
+        assertEquals(APPLICATION_PROBLEM_JSON, response.getHeaders().get(HttpHeaders.CONTENT_TYPE));
+    }
+
+    @Test
+    void handlerThrowsBadGatewayExceptionWhenThereIsThrownIoExceptionWhenReadingFromJson() throws Exception {
+        cristinApiClientStub = spy(cristinApiClientStub);
+
+        doReturn(new HttpResponseStub(IoUtils.stringToStream("")))
+            .when(cristinApiClientStub).fetchQueryResults(any(URI.class));
+        handler = new FetchCristinProjects(cristinApiClientStub, environment);
+        GatewayResponse<ProjectsWrapper> response = sendDefaultQuery();
+
+        assertEquals(HttpURLConnection.HTTP_BAD_GATEWAY, response.getStatusCode());
+        assertEquals(APPLICATION_PROBLEM_JSON, response.getHeaders().get(HttpHeaders.CONTENT_TYPE));
+    }
+
+    @Test
+    void handlerReturnsInternalErrorWhenUriCreationFails() throws Exception {
+        cristinApiClientStub = spy(cristinApiClientStub);
+
+        doThrow(URISyntaxException.class).when(cristinApiClientStub).generateQueryProjectsUrl(any());
+
+        handler = new FetchCristinProjects(cristinApiClientStub, environment);
+        GatewayResponse<ProjectsWrapper> response = sendDefaultQuery();
+
+        assertEquals(HttpURLConnection.HTTP_INTERNAL_ERROR, response.getStatusCode());
+        assertEquals(APPLICATION_PROBLEM_JSON, response.getHeaders().get(HttpHeaders.CONTENT_TYPE));
+    }
+
+    @Test
+    void cristinQueryClassBuildsCorrectUriWhenSuppliedWithParameters() throws Exception {
```

Comment:
This belongs to the CristinQueryTest class, and the test name should follow the pattern.

Reply:
Can you elaborate on test name? Something like `buildsCorrectQueryUriWithParametersWhenUsingCristinQueryClass`?

Reply:
buildRetursnUriWithCustomParameterValuesWhenCustomParameterValuesAreSupplied